### PR TITLE
Validate the exp and nbf JWT claims

### PR DESCRIPTION
**Pull request by davidsansome**

### CODE REVIEW

File: pkg/dss/auth/auth.go [hunk 1]
```diff
@@ -84,21 +92,29 @@ func (a *authClient) AuthInterceptor(ctx context.Context, req interface{}, info
...
+	_, err := jwt.ParseWithClaims(tknStr, &c, func(token *jwt.Token) (interface{}, error) {
 		return a.key, nil
 	})
 	if err != nil {
 		fmt.Println(err)
 		return nil, dsserr.Unauthenticated("invalid token")
 	}
 
-	if err := a.missingScopes(info, strings.Split(claims.ScopeString, " ")); err != nil {
+	if !c.VerifyExpiresAt(a.clock.Now().Unix(), true) {
```
Comment (steeling) [inline pkg/dss/auth/auth.go]: nit: These get called in the StandardClaim's Valid() method, except we're overriding it.

If in the overriden method we replace the `return nil` with:
`return c.StandardClaims.Valid()` we'll get these call for free*

except it looks like it calls these methods with false as the 2nd param, so if we want to ensure that they are provided then we can keep as is. 

Defer to your judgement here

File: pkg/dss/auth/auth.go [hunk 2]
```diff
@@ -55,7 +57,10 @@ func NewSymmetricAuthClient(keyFile string) (*authClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &authClient{key: bytes}, nil
+	return &authClient{
+		key:   bytes,
+		clock: clockwork.NewRealClock(),
```
Comment (steeling) [inline pkg/dss/auth/auth.go]: any benefit to using clockwork? We should be able to spoof time just as easily with the standard lib time.Time.

File: pkg/dss/auth/auth.go (hunk 1 above)
Comment (davidsansome) [reply pkg/dss/auth/auth.go]: I think required = false should be ok.  There's nothing in the spec that requires these tokens to expire (I think), it's just our token provider's implementation that decides that.

I've changed Valid() to call the StandardClaim's method.

File: pkg/dss/auth/auth.go (hunk 2 above)
Comment (davidsansome) [reply pkg/dss/auth/auth.go]: I noticed the jwt package has a TimeFunc - I've used that instead

**Pull request merged**